### DISCUSSION TITLE
Refresh token not coming through.

### DISCUSSION THREAD

**Discussion by wmckenziepyt** [O365/python-o365#959] [Q&A]
here is my script

from O365 import Account,MSOffice365Protocol,MSGraphProtocol
credentials = ('cientgobldeeguk','secretgobldeeguk')

scopeList = ['offline_access','https://outlook.office.com/Mail.Read',
	     'https://outlook.office.com/User.Read',
	     'https://outlook.office.com/Mail.Send',
	     'https://outlook.office.com/Mail.Read.Shared',
             'https://outlook.office.com/Mail.Send.Shared','https://outlook.office.com/email']

    
protocol = MSOffice365Protocol(default_resource='COI2@hireright.com') 

account = Account(credentials=credentials, protocol=protocol )
if account.authenticate(scopes = scopeList):
    print('Authenticated!')

we have offline access enable in graph on azure as well. Thanks in advance.

**Comment by alejcas** [comment]
I don't understand your problem

**Reply by wmckenziepyt** [reply]
I am only getting the authorization token. There is not a refresh token so I have to keep logging back in every hour.

**Reply by wmckenziepyt** [reply]
Was this not enough data?

**Reply by wmckenziepyt** [reply]
Resolved
